Add tests for app/__init__.py + boy scouting

adoublebarrel opened the pull request:
### What is the context of this PR?
[Trello Card](https://trello.com/c/Nm0Hy1Z3/1041-code-coverage-on-app-init-py)

Aim is to improve code coverage of app/__init__.py. New tests exercise existing code/behaviour that isn't already covered by existing integration tests. I've included comments identifying the behaviour that *isn't* tested and where they seem to be covered already.

Have also removed the Markdown plugin as it's no longer used. Fixed a
typo in a test filename.

### How to review 
- Test file can be found in tests/integration/test_app_create.py
- All tests should pass

Commit:
Add tests for app/__init__.py + boy scouting

New tests exercise existing code/behaviour that isn't already covered
by existing integration tests.

Also removed the Markdown plugin as it's no longer used. Fixed a
typo in a test filename.

jonnyshaw89 reviewed (commented):
Would be good to add a test for `get_minimized_asset` so that we can get 100% coverage on `__init__.py`

collisdigital commented:
I think we might have an issue there as well John! https://github.com/ONSdigital/eq-survey-runner/issues/833

adoublebarrel commented:
Good point, I've added a unit test to cover get_minimized_asset's behaviour. I've also added a couple of direct tests of versioned_url_for.

I added a comment addressing #883 as well.

collisdigital reviewed (approved):
Looks good

jonnyshaw89 commented on tests/integration/test_app_create.py:
```diff
@@ -0,0 +1,165 @@
+    def test_adds_blueprints(self):
+        self.assertGreater(len(app.create_app().blueprints), 0)
+
+    # covered by session/test_login.py
```
Should these commented tests be removed?

adoublebarrel replied:
I left them in because I thought they *might* be useful to anyone who comes after who might wonder why functionality X of create_app() isn't covered in this file.

I'm happy to remove them though, or I could remove the defs and just keep one line comments highlighting where the behaviour is tested?
